Sources/PackageLoading/PackageDescription4Loader.swift

```diff
@@ -298,51 +305,79 @@ extension PackageDependencyDescription.Requirement {
             let identifier = try json.get(String.self, forKey: "identifier")
             self = .exact(Version(string: identifier)!)
 
-        case "localPackage":
-            self = .localPackage
-
         default:
             throw InternalError("invalid dependency \(type)")
         }
     }
 }
 
 extension PackageDependencyDescription {
-    fileprivate init(v4 json: JSON, toolsVersion: ToolsVersion, packageLocation: String, fileSystem: FileSystem) throws {
-        let isBaseURLRemote = URL.scheme(packageLocation) != nil
-
-        func fixLocation(_ location: String, requirement: Requirement) throws -> String {
+    fileprivate init(
+        v4 json: JSON,
+        toolsVersion: ToolsVersion,
+        packageLocation: String,
+        identityResolver: IdentityResolver,
+        fileSystem: FileSystem
+    ) throws {
+        let filePrefix = "file://"
+
+        func fixLocation(_ dependencyLocation: String) throws -> String {
             // If base URL is remote (http/ssh), we can't do any "fixing".
-            if isBaseURLRemote {
-                return location
+            if URL.scheme(packageLocation) != nil {
+                return dependencyLocation
             }
 
-            // If the dependency URL starts with '~/', try to expand it.
-            if location.hasPrefix("~/") {
-                return fileSystem.homeDirectory.appending(RelativePath(String(location.dropFirst(2)))).pathString
-            }
-
-            // If the dependency URL is not remote, try to "fix" it.
-            if URL.scheme(location) == nil {
+            if dependencyLocation.hasPrefix("~/") {
+                // If the dependency URL starts with '~/', try to expand it.
+                return fileSystem.homeDirectory.appending(RelativePath(String(dependencyLocation.dropFirst(2)))).pathString
+            } else if dependencyLocation.hasPrefix(filePrefix) {
+                // FIXME: SwiftPM can't handle file locations with file:// scheme so we need to
```

Comment:
Right, in particular, `file://` is treated as a remote package whose repo just happens to be in the local file system.